modules/dnn/src/net_impl.cpp

```diff
@@ -535,7 +535,6 @@ void Net::Impl::allocateLayer(int lid, const LayersShapesMap& layersShapes)
             inps[i] = *ld.inputBlobs[i];
         }
         layerPtr->finalize(inps, ld.outputBlobs);
-        layerPtr->preferableTarget = preferableTarget;
```

Comment:
Moved it to line 590, because this line runs after type inferense and line 590 runs before type inference